### PR TITLE
Docker Basic Tests Implementation

### PR DESCRIPTION
**Pull request by jkopriva**
Docker Basic Test Implementation includes:

1. Perspective switch
2. Create connection 
3. Pull image 
4. Run container 
5. Check Docker Image Tab 
6. Check Container Tab 
7. Clean after test

### CODE REVIEW

**Comment by vpakan** [comment]
Can one of the admins verify this patch?

**Comment by psrna** [comment]
test this please

**Comment by psrna** [comment]
Hi Josef, 

please squash your commits into one single commit with appropriate description and rebase on current master.

Please add copyright headers to your files.

**Comment by psrna** [comment]
Your plugins should be added to modules (in plugins/pom.xml) and site/category.xml.

**Comment by psrna** [comment]
There are maven build errors, please fix.

**Comment by jkopriva** [comment]
Test this please

**Comment by psrna** [comment]
test this please

File: plugins/org.jboss.tools.docker.reddeer/src/org/jboss/tools/docker/reddeer/core/resources/AbstractConnection.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+	private boolean natureMatches(String connectionName, String[] requiredNatureIds) {
+//		List<String> connectionNatureIds = org.jboss.reddeer.direct. .project.Project.getProjectNatureIds(connectionName);
+//		if (requiredNatureIds != null) {
+//			for (String requiredNatureId: requiredNatureIds) {
+//				if (!projectNatureIds.contains(requiredNatureId)) {
+//					return false;
+//				}
+//			}
+//		}
+		return true;
```
Comment (psrna) [inline plugins/org.jboss.tools.docker.reddeer/src/org/jboss/tools/docker/reddeer/core/resources/AbstractConnection.java]: What is the purpose of this matcher if it returns always true?

File: plugins/org.jboss.tools.docker.reddeer/src/org/jboss/tools/docker/reddeer/core/ui/wizards/DockerConnectionWizard.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+	}
+	
+	@Override
+	public void finish() {
+		String shell = new DefaultShell().getText();
+		//Button testConnectionButton = new PushButton("Test Connection");
+		//testConnectionButton.click();
+		//new WaitUntil(new ShellWithTextIsAvailable("Success"));
+		//Button okButton = new PushButton("OK");
+		//okButton.click();
```
Comment (psrna) [inline plugins/org.jboss.tools.docker.reddeer/src/org/jboss/tools/docker/reddeer/core/ui/wizards/DockerConnectionWizard.java]: please remove code that is not used

File: plugins/org.jboss.tools.docker.reddeer/src/org/jboss/tools/docker/reddeer/ui/ConnectionItem.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+	protected final Logger log = Logger.getLogger(ProjectItem.class);
+	
+	private Connection connection;
+	private String[] path;
+	
+	public ConnectionItem(TreeItem treeItem) {
+		super(treeItem);
+		//TODO
+		//connection =
```
Comment (psrna) [inline plugins/org.jboss.tools.docker.reddeer/src/org/jboss/tools/docker/reddeer/ui/ConnectionItem.java]: Todo?

File: tests/org.jboss.tools.docker.ui.bot.test/src/org/jboss/tools/docker/ui/bot/test/AbstractDockerBotTest.java [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+public abstract class AbstractDockerBotTest {
+//	private static final Logger log = Logger.getLogger(AbstractDockerBotTest.class);
+//	private static LogView errorLogView;
+	
+	@BeforeClass 
+	public static void beforeClass(){
+		new WorkbenchShell().maximize();
+//		errorLogView = new LogView();
+//		errorLogView.open();
```
Comment (psrna) [inline tests/org.jboss.tools.docker.ui.bot.test/src/org/jboss/tools/docker/ui/bot/test/AbstractDockerBotTest.java]: please remove code that is not used

**Comment by jkopriva** [comment]
test this please

**Comment by jkopriva** [comment]
test this please

**Comment by psrna** [comment]
test this please

**Pull request merged**